### DISCUSSION TITLE
Making snapshots without host application for swift package

### DISCUSSION THREAD

**Discussion by agruchala** [pointfreeco/swift-snapshot-testing#725] [Q&A]
I added snapshot testing to swift package test target, package tests run without host application. On snapshots I see background but I can't make them render text. Any ideas how to make it happen? I'm testing SwiftUI views.

**Comment by DandyLyons** [comment]
It's my understanding that this is a limitation of Swift Packages, not SnapshotTesting. This is because Swift Packages cannot contain an entire application. Applications require an Xcode project, and Xcode projects cannot be held inside a Swift Package. 

https://forums.swift.org/t/is-it-possible-to-target-build-and-run-an-ios-swiftui-app-from-inside-of-a-swift-package-sdk/55524/2

Snapshots of individual SwiftUI Views can be made in a Swift Package, but if your snapshot requires a host application then you need to make it in an Xcode project. (Also, when making a snapshot that requires an application, don't forget to set drawHierarchyInKeyWindow to true. See https://github.com/pointfreeco/swift-snapshot-testing/blob/d1082c48edaee17914556517b29cea7e3388b2b7/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift#L29)
